packages/flutter/lib/src/material/segmented_button.dart

```diff
@@ -417,6 +541,88 @@ class SegmentedButtonState<T> extends State<SegmentedButton<T>> {
     super.dispose();
   }
 }
+
+@immutable
+class _SegmentedButtonDefaultOverlay extends MaterialStateProperty<Color?> with Diagnosticable {
+  _SegmentedButtonDefaultOverlay(this.overlay);
+
+  final Color overlay;
+
+  @override
+  Color? resolve(Set<MaterialState> states) {
+    if (states.contains(MaterialState.pressed)) {
+      return overlay.withOpacity(0.24);
+    }
+    if (states.contains(MaterialState.hovered)) {
+      return overlay.withOpacity(0.08);
+    }
+    if (states.contains(MaterialState.focused)) {
+      return overlay.withOpacity(0.24);
+    }
+    return null;
+  }
+}
```

Comment:
This opacity values don't match the M3 default overlay opacity value. See `_SegmentedButtonDefaultsM3` class

Reply:
Looks like this is pending